File: ffupdater/src/main/java/de/marmaro/krt/ffupdater/app/AppList.java
package de.marmaro.krt.ffupdater.app;

import de.marmaro.krt.ffupdater.app.impl.Brave;
import de.marmaro.krt.ffupdater.app.impl.FirefoxBeta;
import de.marmaro.krt.ffupdater.app.impl.FirefoxFocus;
import de.marmaro.krt.ffupdater.app.impl.FirefoxKlar;
import de.marmaro.krt.ffupdater.app.impl.FirefoxLite;
import de.marmaro.krt.ffupdater.app.impl.FirefoxNightly;
import de.marmaro.krt.ffupdater.app.impl.FirefoxRelease;
import de.marmaro.krt.ffupdater.app.impl.Iceraven;
import de.marmaro.krt.ffupdater.app.impl.Lockwise;

public enum AppList {
    FIREFOX_RELEASE(new FirefoxRelease()),
    FIREFOX_BETA(new FirefoxBeta()),
    FIREFOX_NIGHTLY(new FirefoxNightly()),
    FIREFOX_FOCUS(new FirefoxFocus()),
    FIREFOX_KLAR(new FirefoxKlar()),
    FIREFOX_LITE(new FirefoxLite()),
    LOCKWISE(new Lockwise()),
    BRAVE(new Brave()),
    ICERAVEN(new Iceraven());

    private final BaseApp baseApp;

    AppList(BaseApp baseApp) {
        this.baseApp = baseApp;
    }

    public BaseApp getBaseApp() {
        return baseApp;
    }
}

File: ffupdater/src/main/java/de/marmaro/krt/ffupdater/app/AppList.kt
package de.marmaro.krt.ffupdater.app

import de.marmaro.krt.ffupdater.app.impl.*
import de.marmaro.krt.ffupdater.app.impl.fetch.ApiConsumer

private val apiConsumer = ApiConsumer()

// TODO ich brauche ihr noch einnen besseren Namen
enum class AppList(impl: App) {
    FIREFOX_RELEASE(FirefoxRelease(apiConsumer)),
    FIREFOX_BETA(FirefoxBeta(apiConsumer)),
    FIREFOX_NIGHTLY(FirefoxNightly(apiConsumer)),
    FIREFOX_FOCUS(FirefoxFocus(apiConsumer)),
    FIREFOX_KLAR(FirefoxKlar(apiConsumer)),
    FIREFOX_LITE(FirefoxLite(apiConsumer)),
    LOCKWISE(Lockwise(apiConsumer)),
    BRAVE(Brave(apiConsumer)),
    ICERAVEN(Iceraven(apiConsumer));

    val impl: App

    init {
        this.impl = CacheWrapper(impl)
    }
}